core/src/main/scala/org/bitcoins/core/crypto/Sign.scala

```diff
@@ -58,3 +59,14 @@ object Sign {
     }, publicKey)
   }
 }
+
+/** A signing interface for [[ExtKey]] */
+trait ExtSign extends Sign {
```

Comment:
I still think that either `ExtSign` should have a `Sign` rather than extending or `ExtPrivateKey` should extend `ECPrivateKey` rather than having one

Reply:
We can revisit this later, as [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#serialization-format) does say that `ExtKey` does have a `BaseECKey`

>33 bytes: the public key or private key data (serP(K) for public keys, 0x00 || ser256(k) for private keys)